source/agora/consensus/PreImage.d

```diff
@@ -487,3 +490,73 @@ unittest
     const number_of_cycles = 12;
     testPreImageCycle(cycle_length, number_of_cycles);
 }
+
+///
+version (unittest) public Hash getSeedAt(in Scalar secret, in ulong total_images,
+    in uint cycle_length, in Height at) nothrow
+{
+    auto cycle = PreImageCycle(secret, cycle_length, total_images, Height(0));
+    return cycle[at];
+}
+
+/// To get the cycle seeds for `PreImageCycle`s of some of well-known keys,
+/// which are used for getting 1000 number of pre-images
+version (unittest) public void getCycleSeed (in KeyPair pair, out Hash seed,
+    out Height seed_height)
+{
+    import agora.utils.WellKnownKeys;
+    import std.format;
+
+    seed_height = Height(999); // which means 1000 number of pre-images.
+    if (pair == NODE7)
+        seed = Hash(`0xee7941fc52af778dcafecd2c4695339b25f0be075b8bff010e0aded69a580a4617f80a3c2c068efa4b2fa7161162142732b15b7b192ac35ffe694e74d0946d0d`);
```

Comment:
What does those hardcoded value correspond to ?

Reply:
It's the seed for the 1000 pre-images for `NODE7` which is in fact the hash value hashed `PreImageCycle.PreImageCount` - 1000 times from the code of `hashMulti(secret, "consensus.preimages", 0)`. Anyway I'll documenting it.

Reply:
Added more detail.